### PR TITLE
Project1 data model

### PR DESCRIPTION
**Pull request by IamYunusAli**
In these pull requests, I have managed to do the following things.

- Create a new app
- Initialize your project with Git and follow git-flow
- Make sure that your project has a Postgres database set up.
- Build your project schema.
    - Create and run the necessary migration files.
    - Table and column names should match the [ERD diagram.](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)
        - note: photo for the user's table should be a link to an image
    - Foreign keys should be included.
    - All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/migrate/20221122111208_add_ref_users_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddRefUsersToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, foreign_key: true
```
Comment (omar-muhamad) [inline db/migrate/20221122111208_add_ref_users_to_posts.rb]: - [ ] Please add the foreign key between users and posts table as `author_id`, not `user_id` as you did now to stick to the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) diagram.
[Hint] You can do it like that:
```ruby
add_reference :posts, :author, foreign_key: { to_table: :users }, null: false
```

File: db/migrate/20221122111235_add_ref_users_and_posts_to_likes.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddRefUsersAndPostsToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, foreign_key: true
```
Comment (omar-muhamad) [inline db/migrate/20221122111235_add_ref_users_and_posts_to_likes.rb]: - [ ] Same here, Please add the foreign key between users and likes table as `author_id`, not `user_id` as you did now to stick to the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) diagram.

File: db/migrate/20221122111322_add_ref_users_and_posts_to_comments.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddRefUsersAndPostsToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, foreign_key: true
```
Comment (omar-muhamad) [inline db/migrate/20221122111322_add_ref_users_and_posts_to_comments.rb]: - [ ] Same here, Please add the foreign key between users and comments table as `author_id`, not `user_id` as you did now to stick to the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) diagram.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @IamYunusAli,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by brytebee — APPROVED** [review]
Hi, @IamYunusAli👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



##

###  **Optional Suggestions:**
- N/A

##

Good luck! 🎉 

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221122111322_add_ref_users_and_posts_to_comments.rb (hunk 3 above)
Comment (IamYunusAli) [reply db/migrate/20221122111322_add_ref_users_and_posts_to_comments.rb]: @Omar-Muhamad thank you for the immense feedback. I have corrected them based on the way you have mentioned. thank you.

**Pull request merged**